src/scipp/plt/mesh.py

```diff
@@ -68,24 +69,26 @@ def _make_mesh(self):
                                          self._data.data.values,
                                          cmap=self._cmap,
                                          shading='auto')
-        self._cbar = plt.colorbar(self._mesh,
-                                  ax=self._ax,
-                                  cax=self._cax,
-                                  extend=self._extend,
-                                  label=name_with_unit(var=self._data.data, name=""))
-
-        # Add event that toggles the norm of the colorbar when clicked on
-        # TODO: change this to a double-click event once this is supported in jupyterlab
-        # see https://github.com/matplotlib/ipympl/pull/446
-        self._cbar.ax.set_picker(5)
-        self._ax.figure.canvas.mpl_connect('pick_event', self.toggle_norm)
-
-        if self._cax is None:
-            self._cbar.ax.yaxis.set_label_coords(-1.1, 0.5)
-        # When we transpose, we remove the mesh and make a new one calling _make_mesh().
-        # To ensure this doesn't add a new colorbar every time we hit transpose, we save
-        # and re-use the colorbar axis.
-        self._cax = self._cbar.ax
+        if self._cbar:
```

Comment:
Oh, I didn't notice that this here is not `__init__`. (Was hidden by GH.) Not sure if this would benefit from a more functional design. You might have to pass a lot of objects around. Do what you think works best.